test_fms/mpp/test_mpp_init_logfile.F90

```diff
@@ -0,0 +1,46 @@
+!***********************************************************************
+!*                   GNU Lesser General Public License
+!*
+!* This file is part of the GFDL Flexible Modeling System (FMS).
+!*
+!* FMS is free software: you can redistribute it and/or modify it under
+!* the terms of the GNU Lesser General Public License as published by
+!* the Free Software Foundation, either version 3 of the License, or (at
+!* your option) any later version.
+!*
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+
+!> @file
+!! @brief Tests the mpp_init_logfile interface.
+!! The files to be replaced are created and checked by the calling
+!! shell program test_mpp_init_logfile.sh.
+!!
+!! @author Miguel Zuniga
+!! @email gfdl.climate.model.info@noaa.gov
+
+program test_mpp_init_logfile
+
+  use mpp_mod, only : mpp_init, mpp_init_logfile
+  use mpp_mod, only : mpp_init_test_logfile_init
+
+  IMPLICIT NONE
+
+  integer :: err_no
+
+  !Initialize mpp but have mpp_init() exit right after log files are initalized.
+ 
+  call mpp_init( test_level =  mpp_init_test_logfile_init -1 )
```

Comment:
The problem that I have with this is that you can't guarantee that it will work in the future.  If they decide to change the test levels to be multiples of 100, switch to reals, etc, then this code will no longer work.  You should just use mpp_test_level instead of subtracting 1

Reply:
I changed the code to use mpp_init_test_logfile_init instead of "mpp_init_test_logfile_init -1"